google/cloud/bigquery/_pandas_helpers.py

```diff
@@ -274,6 +286,22 @@ def types_mapper(arrow_data_type):
         elif time_dtype is not None and pyarrow.types.is_time(arrow_data_type):
             return time_dtype
 
+        elif pyarrow.types.is_struct(arrow_data_type):
```

Comment:
Do we need to handle structs more generally here, or is that logic elsewhere?

Reply:
Good question! Indeed, our types mapper function doesn't seem to do any conversion for STRUCT or ARRAY. This function is used as the parameter `types_mapper` to [pyarrow's `Table.to_pandas()`](https://arrow.apache.org/docs/python/generated/pyarrow.Table.html#pyarrow.Table.to_pandas), allowing for customizable type mapping from pyarrow to pandas. I'm not entirely sure why we didn't provide struct/array mapping options, but I think it might be related to the fact that the fields of a struct can be of any type, so the conversion can become quite complicated.